HomeShopWork/src/сome/shop/work/model/Shop.java

```diff
@@ -0,0 +1,57 @@
+package сome.shop.work.model;
+
+import java.util.*;
+
+public class Shop {
+
+    private Set<Goods> products;
+
+    public Shop() {
+
+        products = new LinkedHashSet<>();
+    }
+
+    public boolean addGood(Goods gds) {
+        return products.add(gds);
+    }
+
+    public Set<Goods> getProducts() {
+        return products;
+    }
+
+    public void setProducts(Set<Goods> products) {
+        this.products = products;
+    }
+
+    public void infoShop() {
+        for (Goods s : products) {
+            if( s == null){
+                System.out.println("Товара нет в магазине!");
+            }
+            System.out.println(s);
+        }
+    }
+
+    public void remove(int id) {
+        for (Goods ce : products) {
```

Comment:
you can replace it with this code, because your equals method of Goods class compares objects by id
Goods good = new Goods();
good.setId(id);
products.remove(good);